cpp/ql/lib/codeql/codeql-pack.release.yml

```diff
@@ -0,0 +1,2 @@
+---
```

Comment:
@dbartol What is the purpose of this file? To me it feels strange to keep this data in the source tree. It's yet another file that will cause conflicts on merge back. It looks like the same value is stored in each of the qlpack files.

Reply:
This stores the version number of the most recently released version of the pack. That will not necessarily be the version that is committed to the qlpack in `main` after the release. The `codeql pack post-release` command (see https://github.com/github/semmle-code/pull/40925) can update the committed version number to have a prerelease suffix so that builds of a package from `main` are not confused with official releases. We already do something similar for the CLI version itself.

Because these files are only updated by the `codeql pack release` command, the only way to have a merge conflict is if a later release was merged back to `main` _before_ a previous release was merged back. This could theoretically happen with a servicing release (say, we released version `2.0.0` of a package, but then discovered a serious issue in `1.5.4` and released a `1.5.5` that fixed to for customers who weren't yet ready to update to `2.0.0`). However, this should be rare enough that manually resolving the conflict in favor of `main` won't be a big deal when it does happen.